New tern version + support for`defs` and `loadEagerly`

iamFIREcracker opened the pull request:
## Update tern.js

The newer version of tern.js ships with a new interface for starting
a server programmatically (https://github.com/ternjs/tern/pull/1000),
and by using that we can simplify the integration a bit.

## Fix the order of arguments of `resolveFrom` call


## Add support for loading tern custom definitions


## Propagate the `loadEagerly` flag

iamFIREcracker committed:
Update tern.js

The newer version of tern.js ships with a new interface for starting
a server programmatically (https://github.com/ternjs/tern/pull/1000),
and by using that we can simplify the integration a bit.

iamFIREcracker committed:
Fix the order of arguments of `resolveFrom` call

iamFIREcracker committed:
Add support for loading tern custom definitions

iamFIREcracker committed:
Propagate the `loadEagerly` flag

tbodt commented on tern-project.js:
```diff
@@ -46,11 +46,22 @@ module.exports = function(root) {
+        }
+        acc.push(file);
+        return acc;
+    }, []);
```
IMO this would be easier to read as a for loop
```suggestion
    const defs = [];
    for (const lib of (project.libs || [])) {
        const file = find('defs', lib, '.json');
        if (file === undefined) {
            // eslint-disable-next-line no-console
            console.error(`Failed to load tern lib ${lib}`);
            continue;
        }
        defs.push(file);
    }
```

iamFIREcracker replied:
Lol, did not notice the `Commit suggestion` button!

Fixed with: df77191

tbodt reviewed (commented):
Thanks for the PR! Just one stylistic comment; also you misspelled `loadEagerly` in your commit message :)

tbodt commented on analysis.js:
```diff
@@ -1,26 +1,14 @@
-// TODO loadEagerly
+const tern = bootstrapServer({
+  projectDir: root,
+  debug: true,
```
Why unconditionally?

iamFIREcracker replied:
No reason; I did turn that on while debugging, but might have forgotten about it.  Will remove this for now.

iamFIREcracker replied:
Fixed with: df77191

iamFIREcracker committed:
Address PR comments

- Use `for-of` loop instead of `reduce`, to improve readability
- Disable debug mode -- accidentally toggled on

tbodt commented:
Thanks again!
